### PR TITLE
Kubectl scale command accepts a filename param

### PR DESCRIPTION
**Pull request by feihujiang**
address part of issue #10628

### CODE REVIEW

**Comment by feihujiang** [comment]
@JanetKuo @caesarxuchao Please review this PR.

**Comment by liggitt** [comment]
@smarterclayton, did you say this filename parameter helper method doesn't work with `-f -`?

**Comment by bgrant0607** [comment]
`-f -` works in general. Is there a reason why it wouldn't work here?

**Comment by smarterclayton** [comment]
I think it was the helper method for registering the param value and
converting it - dash isn't a valid URL maybe?

On Aug 11, 2015, at 12:46 PM, Brian Grant <notifications@github.com> wrote:

-f - works in general. Is there a reason why it wouldn't work here?

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12521#issuecomment-129963211>
.

File: pkg/kubectl/cmd/scale.go [hunk 1]
```diff
@@ -39,6 +39,9 @@ scale is sent to the server.`
 	scale_example = `// Scale replication controller named 'foo' to 3.
 $ kubectl scale --replicas=3 replicationcontrollers foo
 
+// Scale replication controller from foo-controller.yaml to 3.
```
Comment (JanetKuo) [inline pkg/kubectl/cmd/scale.go]: Same comment as other kubectl reviews.

File: pkg/kubectl/cmd/scale.go [hunk 2]
```diff
@@ -68,6 +71,9 @@ func NewCmdScale(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().Duration("timeout", 0, "The length of time to wait before giving up on a scale operation, zero means don't wait.")
 	cmd.MarkFlagRequired("replicas")
 	cmdutil.AddOutputFlagsForMutation(cmd)
+
+	usage := "Filename, directory, or URL to a file containing the replication controller to set a new size"
```
Comment (JanetKuo) [inline pkg/kubectl/cmd/scale.go]: ditto

File: pkg/kubectl/cmd/scale.go [hunk 3]
```diff
@@ -98,16 +105,13 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
...
-		return err
-	}
 
 	infos, err := r.Infos()
 	if err != nil {
 		return err
 	}
-
+	info := infos[0]
+	mapping := info.ResourceMapping()
```
Comment (JanetKuo) [inline pkg/kubectl/cmd/scale.go]: What's the reason behind this change?

**Comment by JanetKuo** [comment]
Nits, otherwise looks good.

**Comment by smarterclayton** [comment]
For sanity can you verify that scale with this change still accepts an object via stdin with -f -?

**Comment by feihujiang** [comment]
@smarterclayton OK, I'll verify this.

File: pkg/kubectl/cmd/scale.go (hunk 3 above)
Comment (feihujiang) [inline pkg/kubectl/cmd/scale.go]: Using the old way of retrieving mapping returned an error "a resource mapping could not be loaded from *resource.FileVisitor".

**Comment by feihujiang** [comment]
@smarterclayton I verified this, and it works. Two tests are as follows:
$ cat ../../../../../examples/cassandra/cassandra-controller.yaml | ./kubectl scale --replicas=1 -f -
replicationcontroller "cassandra" scaled

$ cat ./examples/cassandra/cassandra-controller.yaml | ./cluster/kubectl.sh  scale --replicas=2 -f -
replicationcontroller "cassandra" scaled

@JanetKuo Please review.

**Comment by JanetKuo** [comment]
Looks good. Thanks!

**Comment by JanetKuo** [comment]
@smarterclayton and @caesarxuchao may have more comments; otherwise I'll put the LGTM label.

**Comment by smarterclayton** [comment]
Great, no other comments from me.  Thanks!
